src/permutations.jl

```diff
@@ -10,66 +10,122 @@ export
 
 
 struct Permutations{T}
-    a::T
-    t::Int
+    data::T
+    length::Int
 end
 
-Base.eltype(::Type{Permutations{T}}) where {T} = Vector{eltype(T)}
+function has_repeats(state::Vector{Int})
+    # This can be safely marked inbounds because of the type restriction in the signature.
+    # If the type restriction is ever loosened, please check safety of the `@inbounds`
+    @inbounds for outer in eachindex(state)
+        @inbounds for inner in (outer+1):lastindex(state)
```

Comment:
```suggestion
        for inner in (outer+1):lastindex(state)
```
`@inbounds` applies to the whole block, so only the outermost use is needed here.